lms/services/application_instance.py

```diff
@@ -131,27 +154,26 @@ def create_application_instance(  # pylint:disable=too-many-arguments
             "Hypothesis" + secrets.token_hex(16) if not deployment_id else None
         )
 
-        if developer_secret and developer_key:
-            aes_iv = self._aes_service.build_iv()
-            encrypted_secret = self._aes_service.encrypt(aes_iv, developer_secret)
-        else:
-            # If either one of developer_key or developer_secret is missing, then we
-            # don't save the other one either.
-            developer_key = encrypted_secret = developer_secret = aes_iv = None
-
         application_instance = ApplicationInstance(
             consumer_key=consumer_key,
             shared_secret=secrets.token_hex(32),
             lms_url=lms_url,
             requesters_email=email,
-            developer_key=developer_key,
-            developer_secret=encrypted_secret,
-            aes_cipher_iv=aes_iv,
             created=datetime.utcnow(),
             settings=settings or {},
             deployment_id=deployment_id,
             lti_registration_id=lti_registration_id,
         )
+        # If either one of developer_key or developer_secret is missing,
+        # then we won't set either on creation
+        if not all([developer_secret, developer_key]):
```

Comment:
On creation it makes sense to keep the original behavior, only set one if both are present.

While upgrading update/set either.